### DISCUSSION TITLE
Decrease memory consumption for project that don't use openpyxl

### DISCUSSION THREAD

**Discussion by SebCorbin** [wagtail/wagtail#12125] [Development]
`SpreadsheetExportMixin` is a wonderful mixin but for projects that don't use XLSX reports, the memory increase done by importing `openpyxl` could be avoided. How do you feel about importing `openpyxl` locally in `SpreadsheetExportMixin.write_xlsx()` and `SpreadsheetExportMixin.generate_xlsx_row()`?

**Comment by RealOrangeOne** [comment]
In principle, I like this idea. XLSX exporting is likely quite niche, and reducing memory usage for everyone is a great idea.

However, `openpyxl` 3.1.3 should improve the memory consumption quite a bit, as it stops pre-computing a bunch of data at import time. I think changing the lower-bound constraint on the version would likely net the same kind of memory usage improvement (although I'd love to see a benchmark between the 2).

See https://foss.heptapod.net/openpyxl/openpyxl/-/issues/1401 for more details on the memory usage issues.

**Reply by SebCorbin** [reply]
This issue seems not very active, so I've made a simple PR that move `openpyxl` imports to local imports https://github.com/wagtail/wagtail/pull/12650
